### PR TITLE
virtio_fs: new case of selinux-label support

### PR DESCRIPTION
**Pull request by xiagao**
Selinux-label option support for virtiofsd.
ID: 2102454
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
--testcase=virtio_fs_security_label.security_xattr_check
 (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.security_xattr_check.q35: STARTED
 (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.security_xattr_check.q35: PASS (95.70 s)

**Comment by xiagao** [comment]
@zhenyzha @YongxueHong Call for your review, thanks.

File: qemu/tests/virtio_fs_share_data.py [hunk 1]
```diff
@@ -592,6 +701,22 @@ def viofs_svc_stop_start(action, cmd, expect_status):
...
                 cmd_del = "rm -rf " + dd_of_on_host
                 process.run(cmd_del, timeout=60)
+        if security_label_test:
+            if not se_mode_host_before == se_mode:
+                try:
+                    utils_selinux.set_status(se_mode_host_before)
+                except Exception as err_msg:
+                    test.log.warn("Restore selinux failed with %s on "
+                                  "host." % str(err_msg))
+            if not se_mode_guest_before == se_mode:
```
Comment (zhenyzha) [inline qemu/tests/virtio_fs_share_data.py:711]:
```
[stdlog] 2022-09-27 02:00:49,015 avocado.test ERROR| Reproduced traceback from: /home/kar/workspace/avocado-vt/avocado_vt/test.py:274
[stdlog] 2022-09-27 02:00:49,016 avocado.test ERROR| Traceback (most recent call last):
[stdlog] 2022-09-27 02:00:49,016 avocado.test ERROR|   File "/home/kar/workspace/avocado-vt/virttest/error_context.py", line 135, in new_fn
[stdlog] 2022-09-27 02:00:49,016 avocado.test ERROR|     return fn(*args, **kwargs)
[stdlog] 2022-09-27 02:00:49,016 avocado.test ERROR|   File "/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/virtio_fs_share_data.py", line 711, in run
[stdlog] 2022-09-27 02:00:49,016 avocado.test ERROR|     if not se_mode_guest_before == se_mode:
[stdlog] 2022-09-27 02:00:49,017 avocado.test ERROR| UnboundLocalError: local variable 'se_mode_guest_before' referenced before assignment
```

File: qemu/tests/virtio_fs_share_data.py [hunk 2]
```diff
@@ -548,6 +569,94 @@ def viofs_svc_stop_start(action, cmd, expect_status):
...
+                                      "Shared folder context type is %s"
+                                      % (context_type_file,
+                                         context_type_folder))
+
+                    test.log.info("Security.selinux xattr check with xattr mapping.")
+                    # make sure selinux is enabled on host
+                    se_mode = "enforcing"
+                    error_context.context("Set selinux to %s status on"
+                                          " guest." % se_mode, test.log.info)
+                    se_mode_guest_before = session.cmd_output("getenforce").strip()
```
Comment (zhenyzha) [inline qemu/tests/virtio_fs_share_data.py:625]: "se_mode_guest_before"  
The assignment of "se_mode_guest_before" is in the presence of "getfattr_cmd". Maybe we should set a default value globally?

Comment (xiagao) [reply qemu/tests/virtio_fs_share_data.py:625]: Make sense, I modified code and pls have a check again.

**Comment by zhenyzha** [comment]
` (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.virtio_fs_security_label.arm64-pci: PASS (98.62 s)`

I see an error message in `virtiofs_fs-virtiofs.log`

`WARN  virtiofsd] Use of deprecated option format '-o': Please specify options without it (e.g., '--cache auto' instead of '-o cache=auto')`

So do we need to add a default cache?

**Comment by xiagao** [comment]
> ` (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.virtio_fs_security_label.arm64-pci: PASS (98.62 s)`
> 
> I see an error message in `virtiofs_fs-virtiofs.log`
> 
> `WARN virtiofsd] Use of deprecated option format '-o': Please specify options without it (e.g., '--cache auto' instead of '-o cache=auto')`
> 
> So do we need to add a default cache?

Thanks, we don't need to add a cache because the default cache is auto if we have no --cache defined.
But you give a hint, I use --xattrmap instead of -o xattrmap.

File: qemu/tests/cfg/virtio_fs_security_label.cfg [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        pre_command = "echo 3 > /proc/sys/vm/drop_caches"
+        setup_hugepages = yes
+        kvm_module_parameters = 'hpage=1'
+        expected_hugepage_size = 1024
+        cmd_dd = 'dd if=/dev/urandom of=%s bs=1M count=200 oflag=direct'
+    !s390, s390x:
+        guest_numa_nodes = shm0
+        numa_memdev_shm0 = mem-mem1
+        numa_nodeid_shm0 = 0
+    Windows:
```
Comment (zhenyzha) [inline qemu/tests/cfg/virtio_fs_security_label.cfg:54]: Since line 3 is set to `no Windows` so I think this part can be removed

Comment (xiagao) [reply qemu/tests/cfg/virtio_fs_security_label.cfg]: Yes, you were right. updated yet.

**Comment by zhenyzha** [comment]
` (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.virtio_fs_security_label.arm64-pci: PASS (60.66 s)`

**Review by zhenyzha — APPROVED** [review]

**Review by YongxueHong — APPROVED** [review]

**Comment by YongxueHong** [comment]
Hi @xiagao 
Please retest it with the latest patch, suggest running it with --repeat=5 in the command line.
Thanks

**Comment by xiagao** [comment]
(1/5) repeat1.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: STARTED
 (1/5) repeat1.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: PASS (103.55 s)
 (2/5) repeat2.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: STARTED
 (2/5) repeat2.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: PASS (104.08 s)
 (3/5) repeat3.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: STARTED
 (3/5) repeat3.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: PASS (103.36 s)
 (4/5) repeat4.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: STARTED
 (4/5) repeat4.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: PASS (105.56 s)
 (5/5) repeat5.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: STARTED
 (5/5) repeat5.Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.virtio_fs_security_label.q35: PASS (104.09 s)

@YongxueHong  The result looks good.

**Pull request merged**